cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/TranslatedCall.qll

```diff
@@ -374,6 +374,35 @@ class TranslatedSideEffects extends TranslatedElement, TTranslatedSideEffects {
   override Function getFunction() { result = expr.getEnclosingFunction() }
 }
 
+class TranslatedStructorCallSideEffects extends TranslatedSideEffects {
+  TranslatedStructorCallSideEffects() { getParent().(TranslatedStructorCall).hasQualifier() }
```

Comment:
If I understand it correctly, this class adds the side effects of the qualifier on a structor call. Other arguments and qualifiers seem have their side effects modelled with `TranslatedSideEffect`, so why do structor qualifiers have to get modelled here instead?

Reply:
Most structor calls don't have an `Expr` for the qualifier, so they won't have any corresponding `TranslatedArgumentSideEffect`. `TranslatedStructorCall` does a similar thing to handle the qualifier operand to the structor call, but is simpler because it doesn't need to generate any additional instructions.